# ========== FUEL ========

define: &max_vel 0.4
define: &max_acc 0.25

sdf_map:
  resolution: 0.1
  map_size_x: 30 # 11
  map_size_y: 30 # 11
  map_size_z: 3
  ground_height: -0.2
  box_min_x: -1 #-5.5
  box_min_y: -15 #-5.5
  box_max_x:  29 #5.5
  box_max_y:  15 #5.5
  box_min_z:  -0.0
  box_max_z: 2

fsm:
  # Replan if you have executed at least this amount of trajectory AND the next frontier is covered.
  # This helps the drone to execute at least something before replanning because we need to keep it moving
  thresh_replan2: 0.2 # [s]

frontier:
  cluster_min: 0 # used to discard frontiers smaller than this value
  cluster_size_xy: 1.5 # used by PCA to split large frontiers 
  cluster_size_z: 1.5
  
  # minimum distance a viewpoint needs to be from the centroid of a frontier to be considered
  # If higher than rmin, it will discard some bad viewpoints
  min_candidate_dist: 0.5  
  
  # how far must a viewpoint candidate be from unknown space to be considered. Like a safety margin.
  # high values can sometimes mean that there is no vpt available and 'search fails'
  min_candidate_clearance: 0.3

  # Angle delta for viewpoint sampling
  # if you have very small narrow spaces, keep this lower, but doesnt affect too much tbh
  candidate_dphi: 0.261  # "$(eval 15 * 3.1415926 / 180.0)"
  
  # how many concentric circles of viewpoints to sample between rmin and rmax 
  candidate_rnum: 3
  # lower radius of viewpoint sampling around a frontier
  candidate_rmin: 0.75 # [m]
  # max radius for viewpoint sampling around a frontier
  candidate_rmax: 1.75 # [m]
  down_sample: 2

  # how many valid frontier voxels does a viewpoint need to see to be considered
  # lower values mean we have a lot of viewpoints even in small places
  # higher values means we are conservative about what a viewpoint sees
  # also depends a lot on your map resolution
  min_visib_num: 20
  
  # % change that must happen for the frontier to call for recompute
  # Low values mean that we moved along very fast across the space, considering even a small changed to be enough to reevaluate
  # high values mean we are more strict. We make a lot of observations before moving. This is sluggish but makes the environment scene more stable
  # it should be tuned such that your observe smooth motions, without stops or fast acclerations
  min_view_finish_fraction: 0.2
